lib/active_merchant/billing/gateways/omise.rb

```diff
@@ -0,0 +1,376 @@
+      #
+      # === Example
+      #  To create a charge on a card
+      #
+      #   purchase(money, creditcard_object)
+      #
+      #  To create a charge on a token
+      #
+      #   purchase(money, nil, { :token_id => token_id, ... })
+      #
+      #  To create a charge on a customer
+      #
+      #   purchase(money, nil, { :customer_id => customer_id })
+
+      def purchase(money, payment, options={})
+        create_post_for_purchase(money, payment, options)
+      end
+
+      # Authorize a charge.
+      #
+      # ==== Parameters
+      #
+      # * <tt>money</tt>   -- An amount of money to charge in Satang.
+      # * <tt>payment</tt> -- The CreditCard
+      # * <tt>options</tt> -- A standard hash options
+
+      def authorize(money, payment, options={})
+        options[:capture] = 'false'
+        create_post_for_purchase(money, payment, options)
+      end
+
+      # Capture or a pre-authorized charge.
+      #
+      # ==== Parameters
+      #
+      # * <tt>money</tt>     -- An amount of money to charge in Satang.
+      # * <tt>charge_id</tt> -- The Charge object identifier.
+      # * <tt>options</tt>   -- A standard hash options.
+
+      def capture(money, charge_id, options={})
+        post = {}
+        add_amount(post, money, options)
+        commit(:post, "charges/#{CGI.escape(charge_id)}/capture", post, options)
+      end
+
+      # Void a charge.
+      #
+      # ==== Parameters
+      #
+      # * <tt>charge_id</tt> -- The Charge object identifier.
+      # * <tt>options</tt>   -- A standard hash options.
+
+      def void(charge_id, options={})
+        MultiResponse.run(:first) do |r|
+          r.process { commit(:get, "charges/#{CGI.escape(charge_id)}") }
+          r.process { refund(r.params['amount'], charge_id, options) }
+        end.responses.last
+      end
+
+      # Refund a charge.
+      #
+      # ==== Parameters
+      #
+      # * <tt>money</tt>     -- An amount of money to charge in Satang.
+      # * <tt>charge_id</tt> -- The Charge object identifier.
+      # * <tt>options</tt>   -- A standard hash options.
+
+      def refund(money, charge_id, options={})
+        options[:amount] = money if money
+        commit(:post, "charges/#{CGI.escape(charge_id)}/refunds", options)
+      end
+
+      # Backward compatible method for refunding a charge.
+      def credit(money, charge_id, options={})
+        ActiveMerchant.deprecated CREDIT_DEPRECATION_MESSAGE
+        refund(money, charge_id, options)
+      end
+
+      # Create a Customer and associated Card.
+      #
+      # ==== Parameters
+      #
+      # * <tt>payment</tt> -- The CreditCard.
+      # * <tt>options</tt> -- A standard hash options.
+      # * use options set_default_card: true to set the default card
+
+      def store(payment, options={})
+        return Response.new(false, "#{payment.errors.full_messages.join('. ')}") if payment.is_a?(CreditCard) and !payment.valid?
```

Comment:
Don't do/check validation here - that should be handled by the calling code, and/or be seen as an error coming back from the API.